wayland-client/src/display.rs

```diff
@@ -35,6 +37,21 @@ pub enum ConnectError {
     InvalidFd,
 }
 
+impl fmt::Display for ConnectError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        use self::ConnectError::*;
+        f.write_str(match self {
+            NoWaylandLib => "the `libwayland-client.so` library can not be found",
+            XdgRuntimeDirNotSet => "the `XDG_RUNTIME_DIR` variable is not set while it should be",
+            NoCompositorListening => "the compositor is not listening",
```

Comment:
I'd prefer something like "could not find a wayland compositor listening", it'd better reflect the way things are under the hood imo.